ionoscloud/data_source_container_registry.go

```diff
@@ -0,0 +1,190 @@
+package ionoscloud
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	cr "github.com/ionos-cloud/sdk-go-container-registry"
+	crService "github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/container-registry"
+	"log"
+	"strings"
+)
+
+func dataSourceContainerRegistry() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceContainerRegistryRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"partial_match": {
+				Type:        schema.TypeBool,
+				Description: "Whether partial matching is allowed or not when using name argument.",
+				Default:     false,
+				Optional:    true,
+			},
+			"location": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"garbage_collection_schedule": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"days": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			"hostname": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"maintenance_window": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"days": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			"storage_usage": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"bytes": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"updated_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func dataSourceContainerRegistryRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(SdkBundle).ContainerClient
+
+	idValue, idOk := d.GetOk("id")
+	nameValue, nameOk := d.GetOk("name")
+	locationValue, locationOk := d.GetOk("location")
+
+	id := idValue.(string)
+	name := nameValue.(string)
+	location := locationValue.(string)
+
+	if idOk && (nameOk || locationOk) {
+		diags := diag.FromErr(errors.New("id and name or location cannot be both specified in the same time"))
```

Comment:
@cristiGuranIonos in the error message it should be: `at the same time`.

Reply:
in a separate commit. needs to be set as a constant

Reply:
@cristiGuranIonos only to clarify, you mean that this will be done in a later commit in which we will also create a constant for this error message?